linearizer.sh

```diff
@@ -0,0 +1,58 @@
+#!/bin/bash
+
+set -e
+
+parent=""
+last_good_rev=""
+start_commit=""
+
+# try to reuse previous run, this is incremental mode
+while read lrev; do
+	notes=$(git log --format=%N -1 $lrev| sed -ne 's/pile-commit: \([0-9a-f]\+\)/\1/p')
+	if [ ! -z "$notes" ]; then
+		start_commit=$notes
+		break
+	fi
+done  <<<$(git rev-list --no-merges linearized)
+
+if [ "$start_commit" == "$(git rev-parse pile)" ]; then
+	echo "Nothing to do."
+	exit 0
+fi
+
+if [ ! -z "$start_commit" ]; then
+	commits="$start_commit..pile"
+else
+	commits="pile"
```

Comment:
If we don't want to spend a lot of time generating ancient history for the linearized branch, I think we could probably quickly generate most of the really early commits of the linearized tree from ci-tags via a sequence of

```
for dii in `seq $START $END`; do
    prev=$(git commit-tree -p $prev -m "DII $dii" $(git show --no-patch --format='%T' intel/DII_$dii)
done
```

and then just use this script to generate the more recent history.  That won't add commits for any pile updates that came in so close together that CI didn't assign them a number, but I'm not sure we really need super-accurate history more than a month or two back?

Reply:
Maybe, but is it worth it? We can always start from a more recent one if we want and the incremental would continue just building the rest on top. Also at ~25s per rev (should be less than that on early days of any repository) - a weekend would be sufficient to bootstrap